### PR TITLE
freeciv: Use proper URLs

### PR DESCRIPTION
**Pull request by cazfi**
- https instead of http
- Use canonical freeciv wiki URL

Also add a TODO -comment about the need to switch from mostly deprecated gtk3-client to gtk3.22-client.

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>

#### Testing the changes
- I tested the changes in this PR: **YES**|**briefly**|**NO**

NO: I don't have void-linux installation do to any testing, but the change does not touch any functionality.

### CODE REVIEW

File: srcpkgs/freeciv/template [hunk 1]
```diff
@@ -10,12 +10,14 @@ makedepends="SDL2_gfx-devel SDL2_image-devel SDL2_mixer-devel SDL2_ttf-devel
...
-homepage="http://freeciv.org/"
-changelog="https://freeciv.fandom.com/wiki/NEWS-${version}"
+homepage="https://www.freeciv.org/"
+changelog="https://www.freeciv.org/wiki/NEWS-${version}"
 distfiles="${SOURCEFORGE_SITE}/${pkgname}/${pkgname}-${version}.tar.xz"
 checksum=13215adc96be9f2894d5f3a12c78b8ebb9ae06ecdab25fe6bb1794f6e6d2b61b
 nopie_files="/usr/bin/freeciv-ruledit"
 
+# TODO: gtk3 should be replaced by better supported gtk3.22 in any
+#       distribution that has gtk+-3.22 or gtk+-3.24 available.
```
Comment (Vaelatern) [inline srcpkgs/freeciv/template:20]: Why?

We have `gtk+3-3.24.34_1`, what is the purpose of this comment?

File: srcpkgs/freeciv/template [hunk 2]
```diff
@@ -10,12 +10,14 @@ makedepends="SDL2_gfx-devel SDL2_image-devel SDL2_mixer-devel SDL2_ttf-devel
 short_desc="Free and Open Source empire-building strategy game - server/data"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
-homepage="http://freeciv.org/"
-changelog="https://freeciv.fandom.com/wiki/NEWS-${version}"
+homepage="https://www.freeciv.org/"
+changelog="https://www.freeciv.org/wiki/NEWS-${version}"
```
Comment (Vaelatern) [inline srcpkgs/freeciv/template:14]: This URL redirects to the fandom.com URL. Why the change?

File: srcpkgs/freeciv/template (hunk 1 above)
Comment (cazfi) [reply srcpkgs/freeciv/template:20]: Freeciv gtk3-client is an older client that requires only gtk+-3.10. Freeciv gtk3.22-client is better supported client that requires gtk+-3.22 or gtk+-3.24. Because you have that gtk+-3.24.34, and have no need for the compatibility with gtk+ < 3.22, you should switch to gtk3.22-client of Freeciv.

File: srcpkgs/freeciv/template (hunk 2 above)
Comment (cazfi) [reply srcpkgs/freeciv/template:14]: "https://www.freeciv.org/wiki/" is the official URL that does not make the assumption of the wiki being located on wikia. Maybe freeciv project won't be using wikia forever (and that URL worked already before migrating wiki to wikia)

File: srcpkgs/freeciv/template (hunk 1 above)
Comment (Vaelatern) [reply srcpkgs/freeciv/template:20]: Ah, so you are proposing

```
-_clients="gtk3,sdl2"
+_clients="gtk3.22,sdl2"
```
?

Comment (cazfi) [reply srcpkgs/freeciv/template:20]: Yes. Likely requires more changes than that to work properly, but that's the core of it.

**Comment by Vaelatern** [comment]
Thanks!

**Comment by Vaelatern** [comment]
I tested the new client briefly. Seemed to work.
